cql3/restrictions/statement_restrictions.cc

```diff
@@ -380,18 +380,24 @@ statement_restrictions::statement_restrictions(data_dictionary::database db,
         _clustering_prefix_restrictions = extract_clustering_prefix_restrictions(*_where, _schema);
         _partition_range_restrictions = extract_partition_range(*_where, _schema);
     }
+    _has_multi_column = find_binop(_clustering_columns_restrictions, expr::is_multi_column);
+  try {
     auto cf = db.find_column_family(schema);
     auto& sim = cf.get_index_manager();
     const expr::allow_local_index allow_local(
             !has_partition_key_unrestricted_components()
             && partition_key_restrictions_is_all_eq());
-    _has_multi_column = find_binop(_clustering_columns_restrictions, expr::is_multi_column);
     _has_queriable_ck_index = clustering_columns_restrictions_have_supporting_index(sim, allow_local)
             && !type.is_delete();
     _has_queriable_pk_index = parition_key_restrictions_have_supporting_index(sim, allow_local)
             && !type.is_delete();
     _has_queriable_regular_index = expr::index_supports_some_column(_nonprimary_key_restrictions, sim, allow_local)
             && !type.is_delete();
+  } catch (data_dictionary::no_such_column_family&) {
+    _has_queriable_ck_index = false;
+    _has_queriable_pk_index = false;
+    _has_queriable_regular_index = false;
+  }
```

Comment:
Yes, this was a quick hack to get this query working. I will pass down a parameter to this method instead. Indeed, as-is, this can cover up real problems.